vendors/nxp/boards/lpc54018iotmodule/ports/wifi/iot_wifi.c

```diff
@@ -307,43 +344,66 @@ WIFIReturnCode_t WIFI_On( void )
     A_STATUS result;
 
     /* Prevent re-initialization. WiFi is aleady on this is successful. */
-    if (g_wifi_is_on)
+    if( g_wifi_is_on )
+    {
         return eWiFiSuccess;
+    }
 
     /* Initialize Wi-Fi shield */
-    result = (A_STATUS)WIFISHIELD_Init();
-    if (A_OK != result)
+    result = ( A_STATUS ) WIFISHIELD_Init();
+
+    if( A_OK != result )
+    {
         return eWiFiFailure;
+    }
+
+    /* Power off the WLAN and wait 500ms */
+    WIFISHIELD_PowerUp( 0 );
+    vTaskDelay( MSEC_TO_TICK( 500 ) );
```

Comment:
Since we're changing their hard-coded delay value, should we also change it to a #define?

Reply:
Good idea, will change.